### ISSUE TITLE
Heketi Ghost Volumes - How to delete them

### ISSUE THREAD

**Issue by vikramkhatri** [heketi/heketi#1321]
I have following in my IBM Cloud Private 2.1.0.3

```
# heketi-cli --version
heketi-cli v5.0.0-5-gb005e0f-release-5

# gluster --version
glusterfs 3.12.1
```

When I deploy a helm chart and if that chart has 12 PVC that it creates - it looks that Heketi is not able to keep up and with a result that some volumes are in perpetual pending state (read-only file system) - the [issue](https://github.com/heketi/heketi/issues/1318) discusses this.

During above process, some of the volumes become ghost (the volume was not created in Gluster but heketi thinks that they are there and there is no path assigned to them.

For example:

```
[root@node01 pvc]# heketi-cli node info 545a7f5f5c7b75cce92bff1b658501cd
 << -- snip -- >
Id:bf69b157684f343d0a1ef0b59f05cfe4   Name:/dev/disk/by-path/pci-0000:00:10.0-scsi-0:0:3:0State:online    Size (GiB):109     Used (GiB):2       Free (GiB):107
 << -- snip -- >
Id:6f49f886ba1f87b8eea9e6bfc75fad5b   Name:/dev/disk/by-path/pci-0000:00:10.0-scsi-
# heketi-cli node info 7463ff58284504c26a87364eba027c12
Id:6f49f886ba1f87b8eea9e6bfc75fad5b   Name:/dev/disk/by-path/pci-0000:00:10.0-scsi-0:0:3:0State:online    Size (GiB):109     Used (GiB):71      Free (GiB):38
 << -- snip -- >
Id:d8288eec80731c996fe256eec663bd1c   Name:/dev/disk/by-path/pci-0000:00:10.0-scsi-0:0:3:0State:online    Size (GiB):109     Used (GiB):2       Free (GiB):107     
```
`node01` shows used space as `2 GB`, `node 02` shows `71 GB` and `node03` shows `2 GB`.

The `heketi-cli topology info` for node02 shows as:

```
Bricks:
				Id:0615f959661654a4d69fd2403f3cf182   Size (GiB):2       Path: 
				Id:063e53d4f96ae878094d038f001239f7   Size (GiB):1       Path: 
				Id:069579bc2a7f123b0272d1eead7dafaa   Size (GiB):1       Path: 
				Id:14331ad483384eeb7dc76dceed99ab01   Size (GiB):1       Path: 
				Id:3429de2a9d186b07e81e39239ee96060   Size (GiB):10      Path: 
				Id:37fd3e27e412641cd17f355f0a158e12   Size (GiB):2       Path: /var/lib/heketi/mounts/vg_6f49f886ba1f87b8eea9e6bfc75fad5b/brick_37fd3e27e412641cd17f355f0a158e12/brick
				Id:4213f13096a8bdfe3a16d3efeee8d9ff   Size (GiB):1       Path: 
				Id:4f2560caca2b948599e99d73e22fc480   Size (GiB):1       Path: 
				Id:52bdf58eec542901e162b73fea470273   Size (GiB):8       Path: 
				Id:5cd2d8346a1cd95a8e976d2abadbcaf8   Size (GiB):10      Path: 
				Id:66b6bf4ac3486d3ce77f73ba0042d729   Size (GiB):1       Path: 
				Id:8946983a5181e7ef95d67a9010a978c8   Size (GiB):1       Path: 
				Id:9f2eea761afd6770aaf01cc44ef35ae8   Size (GiB):2       Path: 
				Id:af8fe993b79e3f4bc349b66aae9b2572   Size (GiB):1       Path: 
				Id:c7db22f773d886c4d48e7092cc6787a9   Size (GiB):1       Path: 
				Id:c8c80d9e62aedca63d342874b3df7f3c   Size (GiB):10      Path: 
				Id:cb768661127b3af5f09186b35a678287   Size (GiB):5       Path: /var/lib/heketi/mounts/vg_6f49f886ba1f87b8eea9e6bfc75fad5b/brick_cb768661127b3af5f09186b35a678287/brick
				Id:cfc81fb99585936130d5ca4d244c84a3   Size (GiB):1       Path: 
				Id:d6801d974649867aa33b00e9f7c373c8   Size (GiB):2       Path: /var/lib/heketi/mounts/vg_6f49f886ba1f87b8eea9e6bfc75fad5b/brick_d6801d974649867aa33b00e9f7c373c8/brick
				Id:e7ff70a468c0a9d24d2a3c56ef8b3ece   Size (GiB):10      Path: 
```
Out of 20 volumes, 17 are ghosts now and occupying space which does not reflect at the gluster level.

```
# gluster volume list
heketidbstorage
vol_cd4f30bfc1a03cd798a25d08c51175d6
vol_f1c80f1e07da48e593380a0a845e7327
```

How do I get rid of these ghost volumes from Heketi? It thinks that 71 GB is used but the real use is only 2 GB.
```
 Size (GiB):109     Used (GiB):71      Free (GiB):38
```

I am sure that the development knows this and what are the recommendations?

Will newer version of Heketi solve this? Which version?

Will that version work with Gluster 3.1.2.1?

Looking for tips / techniques / workarounds in the current release until we upgrade to the right version of Heketi?

Is there a curl command using your REST API  that can delete these ghost volumes?

I am just looking for a way to delete these ghost volumes.
 
Thank you so much.

**Comment by phlogistonjohn** [comment]
Hi @vikramkhatri if the volumes exist on the heketi side but not on the gluster side, could you please test out if the recently released Heketi v8.0.0 version can now delete the volumes from Heketi?

If you can't upgrade you can try editing the db by exporting it to json, editing the json, and then re-importing it but this is a riskier option that can lead to internal db inconsistencies if you are not familiar with the internals of heketi.

**Comment by phlogistonjohn** [comment]
Sounds like it's safe to close this issue then. Please reopen the issue if encounter the problem again.
